### PR TITLE
Added warning log if issue link type not present

### PR DESCRIPTION
**Pull request by SageTheWizard**
This error correction created problems for me, resulting in numerous issues to have inverted links (i.e. Blocks / Is Blocked By was inverted). 
Debugging the issue was difficult and required to look into this code to determine what actually happened, if a log message was present, it would have simplified my debugging, so I've added it!

### CODE REVIEW

File: jira/client.py [hunk 1]
```diff
@@ -2524,6 +2524,9 @@ def create_issue_link(
         issue_link_types = self.issue_link_types()
 
         if type not in issue_link_types:
+            LOG.warning(
```
Comment (adehad) [inline jira/client.py:2527]:
```suggestion
            self.log.warning(
```

We use this logger object !

**Review by adehad — CHANGES_REQUESTED** [review]
Small change, glad you got it sorted.

**Comment by studioj** [comment]
master seems ok hopefully lets rebase and see

**Review by studioj — APPROVED** [review]

**Comment by studioj** [comment]
> Small change, glad you got it sorted.

changes?

**Comment by SageTheWizard** [comment]
I see there's a failure deploying to the cloud, if this is something caused by my changes please let me know so that I can change as needed.  (I'm doubtful as it's just a log print though :) )

**Review by adehad — APPROVED** [review]
Thanks for this contribution!

**Pull request merged**